src/main/java/slimeknights/tconstruct/plugin/jei/casting/AbstractCastingCategory.java

```diff
@@ -0,0 +1,103 @@
+package slimeknights.tconstruct.plugin.jei.casting;
+
+import mezz.jei.api.constants.VanillaTypes;
+import mezz.jei.api.gui.IRecipeLayout;
+import mezz.jei.api.gui.drawable.IDrawable;
+import mezz.jei.api.gui.drawable.IDrawableAnimated;
+import mezz.jei.api.gui.ingredient.IGuiFluidStackGroup;
+import mezz.jei.api.gui.ingredient.IGuiItemStackGroup;
+import mezz.jei.api.helpers.IGuiHelper;
+import mezz.jei.api.ingredients.IIngredients;
+import mezz.jei.api.recipe.category.IRecipeCategory;
+import net.minecraft.block.Block;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.fluids.FluidStack;
+import slimeknights.tconstruct.library.Util;
+import slimeknights.tconstruct.smeltery.recipe.AbstractCastingRecipe;
+
+import java.awt.*;
```

Comment:
Not a fan of wildcard imports, makes things harder to trace without an IDE.

Reply:
Fair enough, but this is how intellij did it 'automagically' and it matches 1.12

Reply:
IDEA can be configured to not use them. And it shouldn't be that way in 1.12, boni never liked wildcard imports either